src/types/typeGuards/literal.ts

```diff
@@ -173,9 +170,7 @@ export function isTrueLiteralType(type: ts.Type): type is TrueLiteralType {
  * `LiteralType` from typescript except that it allows for it to work on arbitrary types.
  * @category Type Types
  */
-export interface UnknownLiteralType extends FreshableIntrinsicType {
-	value: unknown;
-}
+export interface UnknownLiteralType extends FreshableIntrinsicType {}
```

Comment:
Note: This would be a breaking change.

Reply:
👍 good timing with #531!

Reply:
@JoshuaKGoldberg Should we remove `UnknownLiteralType` or keep it as an alias of `FreshableIntrinsicType` seems we are no longer adding any properties to it?

Reply:
@RebeccaStevens 

> Note: This would be a breaking change.

Coming back to this, I guess I want to gently push back on this. The current type `BooleanLiteralType` (and its subtypes),

```ts
export interface BooleanLiteralType extends UnknownLiteralType {
	intrinsicName: "false" | "true";
	value: boolean;
}
```

does not describe a real runtime object from which we're removing a runtime field. Instead, it is an _inaccurate_ description of a third party runtime value generated by TS. Therefore, any code that currently relies on this field is _already_ broken. Removing this field from the type definition would, as far as I can tell, be strictly beneficial to TS users, since it would reveal the existing runtime bug at type-checking time. The really nice thing about correcting a type is that we know we're not introducing any runtime regressions, since types have no effect at runtime. Thoughts?

***

@JoshuaKGoldberg 


> Actually, come to think of it, maybe we don't need to get rid of them at all - maybe we can just switch from the `value: ...` property to using a literal for the existing `intrinsicName` property? As in:
> 
> https://typescript-eslint.io/play#ts=5.6.2&showAST=types&fileType=.tsx&code=CYUwxgNghgTiAEYD2A7AzgF3gNyhAriAEJJIQhQoBc8ARqeZQNwBQLoksCy6WuBIAKooA1iiQB3avHyjxU1kA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false
> 
> WDYT?

I'm wondering if there's a misunderstanding here? The type `UnknownLiteralType` is not describing the TS `unknown` type; it's a base type for actual literal types, such as `BooleanLiteralType` type (in fact maybe this is the only one??), to extend. Therefore, the intrinsic name is not the literal `'unknown'`. In fact, that type no longer adds any information over its base type (which already includes `intrinsicName: string`), as both @RebeccaStevens and [@typescript-eslint/no-empty-interface](https://typescript-eslint.io/rules/no-empty-interface/) rightly point out. 

So, in theory, based on that information alone, I would say it can be removed.... However, we run into a bit of an awkward situation, since both `isUnknownLiteralType` and `isFreshableIntrinsicType ` are exported, and they have different runtime implementations.

https://github.com/JoshuaKGoldberg/ts-api-utils/blob/ad63aca2cf88ded4afd011c6104720f42c54d7f7/src/types/typeGuards/literal.ts#L169-L191

https://github.com/JoshuaKGoldberg/ts-api-utils/blob/8c747294f24d0adca8817f4028832b855d907b5b/src/types/typeGuards/compound.ts#L7-L30

If we remove `UnknownLiteralType`, and replace its usages with `FreshableIntrinsicType`, then, we have two different type guards for the exact same type, and, well, deciding how to handle that (and avoiding breaking changes)  involves some decision-making that's not really my place to make. Therefore, my approach was to just leave the now-"useless" `UnknownLiteralType` in place for this PR, and see if it came up in review 🤷. I don't know what the right thing to do here is, and/or whether it makes sense to punt to a followup?

Reply:
I am swayed and am 👍 on treating this as a bugfix rather than a breaking change.

Reply:
@joshuakgoldberg what do you think about the `UnknownLiteralType` part?

Reply:
Oh, right. I think it makes sense to punt.

Reply:
> I am swayed and am 👍 on treating this as a bugfix rather than a breaking change.

My reasoning for this being a breaking change is simply that if someone was relying on `value` existing on `UnknownLiteralType` then things would break. If they were using `BooleanLiteralType`, then obviously they have a bug but if they aren't, then their would be nothing wrong with their code that all of a sudden it would no longer work.

This is on of those things though where it just depends on where you draw the line for what counts as a breaking change and what doesn't. Theoretically, all changes are breaking changes as a user may be relying on the thing that was changed.

R.E. [Breaking Change: The CPU no longer overheats when you hold down spacebar](https://xkcd.com/1172/)

Reply:
> My reasoning for this being a breaking change is simply that if someone was relying on `value` existing on `UnknownLiteralType` then things would break. If they were using `BooleanLiteralType`, then obviously they have a bug but if they aren't, then their would be nothing wrong with their code that all of a sudden it would no longer work.

Oh, I missed that the break you were referring to was specifically `UnknownLiteralType` rather than the removal of `value` in general from `BooleanLiteralType` and its subtypes. Not necessarily sure what to do about that (or whether it matters??) but an alternative would be to have `BooleanLiteralType` extend `FreshableIntrinsicType` directly and cut out the `UnknownLiteralType` which adds the erroneous `value` field?

That being said, it still seems like it's a bug to have `value: unknown` on `UnknownLiteralType` since `isUnknownLiteralType` returns `true` for booleans, even though they don't have the `value` field. Maybe it can still be removed? Or maybe it should be changed to `value?: unknown`?

Reply:
I think `value?: unknown` should be fine to use for a bug fix and we can deprecate `UnknownLiteralType`.

In v2 we can remove `UnknownLiteralType` altogether.